docs/src/pages/components/buttons/IconLabelButtons.js

```diff
@@ -1,54 +1,74 @@
 import React from 'react';
-import clsx from 'clsx';
 import Button from '@material-ui/core/Button';
 import { makeStyles } from '@material-ui/core/styles';
 import DeleteIcon from '@material-ui/icons/Delete';
 import CloudUploadIcon from '@material-ui/icons/CloudUpload';
 import KeyboardVoiceIcon from '@material-ui/icons/KeyboardVoice';
 import Icon from '@material-ui/core/Icon';
 import SaveIcon from '@material-ui/icons/Save';
+import CircularProgress from '@material-ui/core/CircularProgress';
 
 const useStyles = makeStyles(theme => ({
   button: {
     margin: theme.spacing(1),
   },
-  leftIcon: {
-    marginRight: theme.spacing(1),
-  },
-  rightIcon: {
-    marginLeft: theme.spacing(1),
-  },
-  iconSmall: {
-    fontSize: 20,
-  },
 }));
 
 export default function IconLabelButtons() {
   const classes = useStyles();
 
   return (
     <div>
-      <Button variant="contained" color="secondary" className={classes.button}>
+      <Button
+        variant="contained"
+        color="secondary"
+        className={classes.button}
+        endIcon={<DeleteIcon fontSize="small" />}
+      >
         Delete
-        <DeleteIcon className={classes.rightIcon} />
       </Button>
-      <Button variant="contained" color="primary" className={classes.button}>
+      {/* This Button uses a Font Icon, see the installation instructions in the Icon component docs. */}
+      <Button
+        variant="contained"
+        color="primary"
+        className={classes.button}
+        endIcon={<Icon fontSize="small">send</Icon>}
+      >
         Send
-        {/* This Button uses a Font Icon, see the installation instructions in the docs. */}
-        <Icon className={classes.rightIcon}>send</Icon>
       </Button>
-      <Button variant="contained" color="default" className={classes.button}>
+      <Button
+        variant="contained"
+        color="default"
+        className={classes.button}
+        endIcon={<CloudUploadIcon fontSize="small" />}
+      >
         Upload
-        <CloudUploadIcon className={classes.rightIcon} />
       </Button>
-      <Button variant="contained" disabled color="secondary" className={classes.button}>
-        <KeyboardVoiceIcon className={classes.leftIcon} />
+      <Button
+        variant="contained"
+        disabled
+        color="secondary"
+        className={classes.button}
+        startIcon={<KeyboardVoiceIcon fontSize="small" />}
+      >
         Talk
       </Button>
-      <Button variant="contained" size="small" className={classes.button}>
-        <SaveIcon className={clsx(classes.leftIcon, classes.iconSmall)} />
+      <Button
+        variant="contained"
+        size="small"
+        className={classes.button}
+        startIcon={<SaveIcon fontSize="small" />}
+      >
         Save
       </Button>
+      <Button
+        variant="contained"
+        color="primary"
+        className={classes.button}
+        startIcon={<CircularProgress color="inherit" size={20} />}
```

Comment:
This is a problem for visual regression tests

Reply:
What do you propose?

Reply:
I would double check that we can't disable the animation when we taken the screenshot. If we can't, we need to remove the demo form the visual tests.

Alternatively, we can defer this problem to when we handle the button loading feature.

Reply:
>  would double check that we can't disable the animation when we taken the screenshot.

I'm sorry, I have no idea how to do that.

> Alternatively, we can defer this problem to when we handle the button loading feature.

I am in two minds about this feature. It's nice to have, but it means that the button imports the CircularProgress component for every button, even if `loading` isn't used. Given how widely used the button is, I'm not sure how wise it is to overload it with features.